lib/config-generator.js

```diff
@@ -148,10 +148,6 @@ const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: plug
         let exportContent = "";
         let needCompatHelper = false;
 
-        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
-            exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
-        }
-
```

Comment:
Move them to the end - avoid being overwritten by 3rd-party configs.

Reply:
I think it could be actually good to let configs overwrite the source type for some files. For example, when a framework expects all files in a particular directory to be in ESM format but the user selects CommonJS. But I don't feel strongly about this and don't have a specific example in mind.

If we decide to overwrite the source type with what the user has selected, then it's better to do it also for `"module"`, not only for `"commonjs"` and `"script"`.

Reply:
right, I reverted the change: 0551e8655418534e82e65a6da7dc31fb7472bf97 :)